cluster/validate-cluster.sh

```diff
@@ -24,23 +24,27 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/cluster/kube-env.sh"
 source "${KUBE_ROOT}/cluster/${KUBERNETES_PROVIDER}/util.sh"
 
+API_VERSION=v1
 MINIONS_FILE=/tmp/minions-$$
 trap 'rm -rf "${MINIONS_FILE}"' EXIT
 
 # Make several attempts to deal with slow cluster birth.
 attempt=0
 while true; do
-  # The "kubectl get nodes" output is three columns like this:
+  # The "kubectl get nodes -o yaml" exports node information in standard yaml format.
   #
-  #     NAME                     LABELS    STATUS
-  #     kubernetes-minion-03nb   <none>    Ready
-  #
-  # Echo the output, strip the first line, then gather 2 counts:
+  # Echo the output and gather 2 counts:
   #  - Total number of nodes.
   #  - Number of "ready" nodes.
-  "${KUBE_ROOT}/cluster/kubectl.sh" get nodes > "${MINIONS_FILE}" || true
-  found=$(cat "${MINIONS_FILE}" | sed '1d' | grep -c .) || true
-  ready=$(cat "${MINIONS_FILE}" | sed '1d' | awk '{print $NF}' | grep -c '^Ready') || true
+  "${KUBE_ROOT}/cluster/kubectl.sh" get nodes --api-version="${API_VERSION}" -o yaml> "${MINIONS_FILE}" || true
+  
+  found=0;ready=0
+  if [[ $API_VERSION == v1 ]]; then
+    # when there is a 'kind: Node' line, there is a node found.
+    found=$(cat "${MINIONS_FILE}" | grep -c 'kind: Node') || true
+    # only when status: "True" and type: Ready both exist in conditions section, the node is "ready".
+    ready=$(cat "${MINIONS_FILE}" | sed -n 'N;/status: "True".*type: Ready/p; D;' | grep -c 'True') || true
```

Comment:
The reason I didn't use `-o template` is when template format changes, it will not working. But as you said, the v1 version may last longer than the script, it should not be a concern. I'll change it to `-o template`